GoogleSignIn/Tests/Unit/GIDRestrictedScopesRegistryTest.m

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+#import "GoogleSignIn/Sources/GIDRestrictedScopesRegistry.h"
+
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h"
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifyAccountDetail.h"
+
+@interface GIDRestrictedScopesRegistryTest : XCTestCase
+@end
+
+@implementation GIDRestrictedScopesRegistryTest
+- (void)testIsScopeRestricted {
+    GIDRestrictedScopesRegistry *registry = [[GIDRestrictedScopesRegistry alloc] init];
+    BOOL isRestricted = [registry isScopeRestricted:kAccountDetailTypeAgeOver18Scope];
+    XCTAssertTrue(isRestricted);
+}
+
+- (void)testRestrictedScopeToClassMappingInSet {
+    GIDRestrictedScopesRegistry *registry = [[GIDRestrictedScopesRegistry alloc] init];
+    NSSet<NSString *> *scopes = [NSSet setWithObjects:kAccountDetailTypeAgeOver18Scope, @"some_other_scope", nil];
```

Comment:
I think you've logically captured this in the tests below, but would you also please add a test ensuring the `"some_other_scope"` is not in the mapping below?

Reply:
Added! Asserted that `"some_other_scope"` isn't in the mapping.